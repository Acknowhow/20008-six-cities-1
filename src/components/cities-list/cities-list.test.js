import React from 'react';
import renderer from 'react-test-renderer';
import CitiesList from './cities-list.jsx';

const mock = {
  names: [`Manila`, `Delhi`, `Abu Dhabi`, `Burkina Faso`, `Kaya`],
  incorrectNames: [0, 1, 4, 5, `New York`]
};

it(`CitiesList renders correctly after launch`, () => {

  const {names} = mock;

  const tree = renderer
    .create(<CitiesList
      activateItem={jest.fn()}
      isActive={jest.fn()}
      cityNames={names}
      handleTabClick={jest.fn()}
    />, {
      createNodeMock: () => {
        return {};
      }})
    .toJSON();

  expect(tree).toMatchSnapshot();

});
