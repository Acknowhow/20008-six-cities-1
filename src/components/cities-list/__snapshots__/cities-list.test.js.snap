// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList renders correctly after launch 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Manila
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Delhi
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Abu Dhabi
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Burkina Faso
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Kaya
      </span>
    </a>
  </li>
</ul>
`;
